### PR TITLE
JAVAFICATION: Ported ShutdownWatcher to Java

### PR DESCRIPTION
**Pull request by original-brownbear**
Exact port of the behaviour of `ShutdownWatcher`.

Note:

* I removed 3 specs
  * One I ported to Java exactly (to make sure the global force shutdown settings still works). Sorry for the inline Ruby code here, it was simply the shortest route to reproducing the existing test.
  * Two of them I dropped. They were testing the internal behaviour of the `ShutdownWatcher`, this is not something that can be tested this way in Java/JUnit, is bad test design and would've taken hours of coding to reproduce in Java for no improvement in the coverage of the outward behaviour of the class.

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins test this

File: logstash-core/src/main/java/org/logstash/execution/ShutdownWatcherExt.java [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+    private static final AtomicBoolean unsafeShutdown = new AtomicBoolean(false);
+
+    private final List<IRubyObject> reports = new ArrayList<>();
+
+    private final AtomicInteger attemptsCount = new AtomicInteger(0);
+
+    private final AtomicBoolean running = new AtomicBoolean(false);
+
+    private long cyclePeriod = TimeUnit.SECONDS.toMillis(5L);
```
Comment (danhermann) [inline logstash-core/src/main/java/org/logstash/execution/ShutdownWatcherExt.java]: Did you intend to change this from the 1sec interval in the Ruby implementation?

Comment (danhermann) [reply logstash-core/src/main/java/org/logstash/execution/ShutdownWatcherExt.java]: Also, won't this definition clash with the later call to `TimeUnit.SECONDS.sleep()`?

Comment (original-brownbear) [reply logstash-core/src/main/java/org/logstash/execution/ShutdownWatcherExt.java]: @danhermann whoops thanks for catching, fixing :)

**Comment by original-brownbear** [comment]
@danhermann fixed the constant to just `1L`, should be exactly like before now :)

**Comment by original-brownbear** [comment]
@danhermann thanks!

**Pull request closed without merging**